### DISCUSSION TITLE
Failover or HA: Ideas?

### DISCUSSION THREAD

**Discussion by 0xD4** [simple-login/app#638] [Self-hosting question] [answered]
Hi

I am considering hosting SimpleLogin myself. Has anyone tried running the system as a failover or HA? I'm trying to figure out what options are available to run the system somewhat reliably and continue to receive and sending mails in the event of a host failure.

Do you have any ideas?

Best,
0

**Comment by nguyenkims** [comment, marked as answer]
@0xD4 the failover and HA is indeed a bit overkill for a personal email server as email protocol is quite fault tolerant: if your server is down, a sender will retry later in the future.

Our self host doesn't support failover out of the box but you can run the same setup on another server and have another MX record pointing to it with a lower priority (i.e. higher priority value in the MX record and yeah this is counterintuitive)  than the main server. Then you would need a shared database that SimpleLogin containers (sl-app and sl-email) can access from these 2 servers.

**Reply by 0xD4** [reply]
Awesome, thank you! Good solution with the second mail server (MX Record). I will think about a concept of how this could work in detail and possibly do some testing...
